.github/ISSUE_TEMPLATE/golang_version_upgrade.md

```diff
@@ -0,0 +1,34 @@
+Steps to upgrade Golang version:
+- [ ] Update `go.mod` and `go.sum`. At the root of the directory, run:
+    - [ ] `find . -name 'go.mod' -not -path '*/\.*' -execdir go mod edit -go=<NEW_GOLANG_VERSION_WITHOUT_PATCH> \;`
+    - [ ] `find . -name 'go.mod' -not -path '*/\.*' -execdir go mod tidy \;`
+
+- [ ] Update the Dockerfiles for `build` directory. At the root of the directory, run: 
+    
+    `find build -type f \( -not -path '*/\.*' -and -not -path 'build/tmp/*' \) -exec sed -i 's/GO_VERSION=[0-9]\+\.[0-9]\+\.[0-9]\+/GO_VERSION=<NEW_GOLANG_VERSION>/g' {} \;`
+    
+- [ ] Update the Dockerfiles for `examples` directory. At the root of the directory, run:     
+    - [ ] `find examples -name Dockerfile -exec sed -i 's/golang:[0-9]\+\.[0-9]\+-alpine/golang:<NEW_GOLANG_VERSION_WITHOUT_PATCH>-alpine/g' {} \;`
+    - [ ] `find examples \( -name Dockerfile -o -name Dockerfile.windows \) -exec sed -i 's/golang:[0-9]\+\.[0-9]\+\.[0-9]\+/golang:<NEW_GOLANG_VERSION>/g' {} \;`
+    
+- [ ] Update the example images tag. At `build` directory, run:
+    - [ ] `make bump-image IMAGENAME=autoscaler-webhook VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=crd-client VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=custom-controller VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=simple-game-server VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=simple-genai-game-server VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=supertuxkart-example VERSION=<current-image-version>`
+    - [ ] `make bump-image IMAGENAME=xonotic-example VERSION=<current-image-version>`
```

Comment:
`allocation-endpoint` is missing from the list. Should we make the instructions more generic?


 Navigate to the `build` directory and use the following command to update each image tag. Replace <image-name> with the specific image name and <current-image-version> with the current version number:
 
 `make bump-image IMAGENAME=<image-name> VERSION=<current-image-version>`

Reply:
I prefer to have more detailed instructions so someone who are not familiar with the code can still follow the steps.
Added instruction for `allocation-endpoint-proxy`

Reply:
> someone who are not familiar with the code can still follow the steps.

That's a good point! Just a thought...  shall we include these steps in a .sh file to automate the upgrade process? But the final call is yours....

```
#!/usr/bin/env bash

read -p "Enter the Golang version : " GOLANG_VERSION

echo "Updating go.mod files..."
find . -name 'go.mod' -not -path '*/\.*' -execdir go mod edit -go=$GOLANG_VERSION \;
find . -name 'go.mod' -not -path '*/\.*' -execdir go mod tidy \;

echo "Updating Dockerfiles in the build directory..."
find build -type f \( -not -path '*/\.*' -and -not -path 'build/tmp/*' \) -exec sed -i "s/GO_VERSION=[0-9]\+\.[0-9]\+\.[0-9]\+/GO_VERSION=$GOLANG_VERSION/g" {} \;

echo "Updating Dockerfiles in the examples directory..."
find examples -name Dockerfile -exec sed -i "s/golang:[0-9]\+\.[0-9]\+-alpine/golang:$GOLANG_VERSION-alpine/g" {} \;
find examples \( -name Dockerfile -o -name Dockerfile.windows \) -exec sed -i "s/golang:[0-9]\+\.[0-9]\+\.[0-9]\+/golang:$GOLANG_VERSION/g" {} \;

cd build

echo "Updating example images tags..."
read -p "Enter the current allocation-endpoint-proxy version: " VERSION_ALLOCATION_ENDPOINT_PROXY
make bump-image IMAGENAME=allocation-endpoint-proxy VERSION=$VERSION_ALLOCATION_ENDPOINT_PROXY

read -p "Enter the current autoscaler-webhook version: " VERSION_AUTOSCALER_WEBHOOK
make bump-image IMAGENAME=autoscaler-webhook VERSION=$VERSION_AUTOSCALER_WEBHOOK

read -p "Enter the current crd-client version: " VERSION_CRD_CLIENT
make bump-image IMAGENAME=crd-client VERSION=$VERSION_CRD_CLIENT

read -p "Enter the current custom-controller version: " VERSION_CUSTOM_CONTROLLER
make bump-image IMAGENAME=custom-controller VERSION=$VERSION_CUSTOM_CONTROLLER

read -p "Enter the current simple-game-server version: " VERSION_SIMPLE_GAME_SERVER
make bump-image IMAGENAME=simple-game-server VERSION=$VERSION_SIMPLE_GAME_SERVER

read -p "Enter the current simple-genai-game-server version: " VERSION_SIMPLE_GENAI_GAME_SERVER
make bump-image IMAGENAME=simple-genai-game-server VERSION=$VERSION_SIMPLE_GENAI_GAME_SERVER

read -p "Enter the current supertuxkart-example version: " VERSION_SUPERTUXKART_EXAMPLE
make bump-image IMAGENAME=supertuxkart-example VERSION=$VERSION_SUPERTUXKART_EXAMPLE

read -p "Enter the current xonotic-example version: " VERSION_XONOTIC_EXAMPLE
make bump-image IMAGENAME=xonotic-example VERSION=$VERSION_XONOTIC_EXAMPLE

```

Reply:
I think given @gongmax's transition, we should just proceed with the checklist - we can automate this in the future, perhaps after the next bump (which will also help validate the checklist). Does that make sense?